src/App.tsx

```diff
@@ -15,47 +16,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+interface SortParams {
+  sortField: SortType;
+  isReversed: boolean;
+}
+
+enum SortType {
+  Default,
+  Alphabet,
+  Length,
+}
+
+function getPreparedGoods(goods: string[],
+  { sortField, isReversed }: SortParams) {
+  const preparedGoods = [...goods];
+
+  if (sortField !== SortType.Default) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.Alphabet:
+          return good1.localeCompare(good2);
+
+        case SortType.Length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.Default);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
+          onClick={() => setSortField(SortType.Alphabet)}
           type="button"
-          className="button is-info is-light"
+          className={
+            cn(
+              'button is-info',
+              { 'is-light': sortField !== SortType.Alphabet },
+            )
+          }
         >
           Sort alphabetically
         </button>
 
         <button
+          onClick={() => setSortField(SortType.Length)}
           type="button"
-          className="button is-success is-light"
+          className={
+            cn(
+              'button is-success',
+              { 'is-light': sortField !== SortType.Length },
+            )
+          }
         >
           Sort by length
         </button>
 
         <button
+          onClick={() => setIsReversed(!isReversed)}
           type="button"
-          className="button is-warning is-light"
+          className={cn(
+            'button is-warning',
+            { 'is-light': !isReversed },
+          )}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(isReversed || sortField !== SortType.Default) && (
+          <button
+            onClick={() => {
+              setSortField(SortType.Default);
+              setIsReversed(false);
+            }}
```

Comment:
Pls move the function inside `onClick` handler to a separate const and use as a callback here